lib/pure/collections/lists.nim

```diff
@@ -7,24 +7,148 @@
 #    distribution, for details about the copyright.
 #
 
-## Implementation of singly and doubly linked lists. Because it makes no sense
-## to do so, the 'next' and 'prev' pointers are not hidden from you and can
+## Implementation of singly and doubly linked lists and rings.
+##
+## Because it makes no sense to do so, the ``next`` and ``prev``
+## pointers are not hidden from you and can
 ## be manipulated directly for efficiency.
+##
+## * ``DoublyLinkedList`` and ``DoublyLinkedRing`` are ``O(1)`` to `remove` elements
+## * ``SinglyLinkedList`` and ``SingleLinkedRing`` are ``O(n)`` to `remove` elements `(with lower memory usage)`
+##
+## Initializing a list/ring
```

Comment:
A lot of the examples below can be significantly simplified. You don't need to show ``toSeq`` for every single type.

Please show ideally single-line/double-line examples instead of this wall of code. People want to skim examples, not understand hundreds of lines of code.